php/libraries/NDB_Config.class.inc

```diff
@@ -92,6 +92,12 @@ class NDB_Config
      */
     function load(string $configFile = "../project/config.xml"): void
     {
+        // in test mode,load configFile from Docker DB
+        $env = getenv('LORIS_DB_CONFIG');
+        if (!empty($env)) {
+            $configFile = $env;
+        }
+
```

Comment:
This is going to override the parameter passed as an argument. The determination of where the config file is done in singleton() (and this also misses a bunch of cases handled there), which should already be taking care of this. Why is this change being added here?

Reply:
This load function will import the Database from /project/config.xml. In the Docker test, this file should be changed read from LORIS_DB_CONFIG. ( docker-compose.yml environment: - LORIS_DB_CONFIG=/app/test/config.xml)

Reply:
The Docker test shouldn't be calling load directly, it should be going through the factory, and it's the caller's responsibility to have called singleton with the correct location the first time.